cockpit/IPC/src/Components/Form/WidgetFunctions.tsx

```diff
@@ -86,16 +104,20 @@ export const updateFormData = (name: string | undefined, newData: any, setFormDa
   });
 };
 
-export async function fetchDataFromDBus(name: string) {
+export async function fetchDataFromDBus(name: string, iface: string, path: string, property: string) {
+  console.log('name = ', name);
+  console.log('iface = ', iface);
   const dbus = window.cockpit.dbus(name);
-  const OBJproxy = dbus.proxy(name, `/${TFC_DBUS_DOMAIN}/${TFC_DBUS_ORGANIZATION}/etc/tfc/config`);
+  console.log('dbus = ', dbus);
+  const OBJproxy = dbus.proxy(iface, `/${TFC_DBUS_DOMAIN}/${TFC_DBUS_ORGANIZATION}/${path}`);
   await OBJproxy.wait();
+  console.log('proxy = ', OBJproxy);
 
   const { data } = OBJproxy;
-  let parsedData = JSON.parse(data.config[0].replace('\\"', '"'));
-  const parsedSchema = JSON.parse(data.config[1].replace('\\"', '"'));
+  let parsedData = JSON.parse(data[property][0].replace('\\"', '"'));
+  const parsedSchema = JSON.parse(data[property][1].replace('\\"', '"'));
```

Comment:
Finnst eins og þetta sé unsafe. geturu athugað lengd before access og sett þá eitthvað log?

Reply:
Added an "exists" check before.
This property is also hard-coded, so it can't fail at runtime unless DBUS changes.